STACpopulator/populator_base.py

```diff
@@ -135,13 +135,50 @@ def create_stac_collection(self) -> dict[str, Any]:
         )
         self._collection_info["extent"] = pystac.Extent(sp_extent, tmp_extent)
         self._collection_info["summaries"] = pystac.Summaries({"needs_summaries_update": ["true"]})
+
+        # Add any assets if provided in the config
+        if "assets" in self._collection_info:
+            self._collection_info["assets"] = self.__make_collection_assets()
+
+        # Construct links if provided in the config
+        if "links" in self._collection_info:
+            collection_links = self.__make_collection_links()
+            # need to remove the links item other constructing a collection object fails
+            _ = self._collection_info.pop("links")
```

Comment:
typo "other" -> "otherwise" ?

I think it would make more sense to delegate all the checks to `self.__make_collection_links` to avoid spliting the logic in two places.
i.e.: 

```python
collection_links = self.__make_collection_links()  # called directly

def __make_collection_links(self):
    links = self._collection_info.pop("links", [])
    for link_name, link_info in self._collection_info["links"].items():
        links.append(pystac.Link(**link_info))
    return links
```

Same idea for `"assets"`.

Reply:
Yeah, that's a good idea. Let me rearrange it that way.